alpakaConfig.cmake

```diff
@@ -213,8 +213,29 @@ IF(NOT Boost_FOUND)
     SET(_ALPAKA_FOUND FALSE)
 
 ELSE()
-    LIST(APPEND _ALPAKA_INCLUDE_DIRECTORIES_PUBLIC ${Boost_INCLUDE_DIRS})
-    LIST(APPEND _ALPAKA_LINK_LIBRARIES_PUBLIC ${Boost_LIBRARIES})
+    IF(Boost_FIBER_FOUND)
+        # Boost fiber and default header-only libraries
+        IF(TARGET Boost::fiber)
+            LIST(APPEND _ALPAKA_LINK_LIBRARIES_PUBLIC
+                 Boost::boost
+                 Boost::fiber Boost::context Boost::system Boost::thread
```

Comment:
As far as I understad the FindBoost script, it already adds the fiber dependencies to the fiber target, so we do not need to mention them here again because they are transitively inherited.

Reply:
Yes, transitive dependencies are pulled in automatically *as long as your CMake is new enough*.

I just checked the implementation of `FindBoost.cmake` again and it luckily changed recently:

- <= 3.9.0: Newer versions have no targets: https://github.com/Kitware/CMake/blob/v3.9.0/Modules/FindBoost.cmake#L752
- 3.10+: Finally! Assume newer releases *just have the same dependencies as before*: https://github.com/Kitware/CMake/blob/v3.10.0/Modules/FindBoost.cmake#L782-L783

The last thing is exactly what I implement here.

I would say, as long as we have our requirement not pushed to CMake 3.10+ let us keep it that explicit for transient dependencies *in case a user has a new Boost version that is not yet known by CMake*.